tensornetwork/network_components.py

```diff
@@ -1843,6 +1908,15 @@ def contract_between(
     new_tensor = backend.tensordot(node1.tensor, node2.tensor, [axes1, axes2])
     new_node = Node(
         tensor=new_tensor, name=name, backend=backend)
+    # Merge and update axis names if nodes use non-generic names
+    if (not axis_names and 
+        (node1.has_nongeneric_axis_names() or 
+         node2.has_nongeneric_axis_names()) and
+        backend.name in ['tensorflow', 'jax', 'numpy', 
+                         'pytorch']):
```

Comment:
Why are you checking backend names here?

Reply:
Yeah I meant to run it by you, I noticed one of the tests failing, specifically `test_custom_backend` in `tensornetwork_test.py`. These axis merging rules will only work on backends implementing normal tensor behavior, and may fail on custom backends, so this was my solution to that failing test. Should I omit this, or is there a better way to check for that?

Reply:
OK yeah was misunderstanding something, fixed now